src/builder/build_traits.rs

```diff
@@ -297,16 +296,15 @@ pub trait Dataflow: Container {
     fn dfg_builder(
         &mut self,
         signature: FunctionType,
-        input_extensions: Option<ExtensionSet>,
         input_wires: impl IntoIterator<Item = Wire>,
     ) -> Result<DFGBuilder<&mut Hugr>, BuildError> {
         let op = ops::DFG {
             signature: signature.clone(),
         };
-        let nodetype = NodeType::new(op, input_extensions.clone());
+        let nodetype = NodeType::new_auto(op);
```

Comment:
The choice of new_auto here is about the only non-obvious thing in the PR. ("pure" for `FuncDefn` but "open" for any other DFG.) I think the next PR might even remove new_auto and that could be a good point to say #702 is finished